words/scanner_test.go

```diff
@@ -381,3 +380,47 @@ func BenchmarkUnicodeSegments(b *testing.B) {
 		b.ReportMetric(float64(c), "tokens")
 	}
 }
+
+func TestScannerMaxTokenLen(t *testing.T) {
+	// Build a string longer than the max token length
+	str := ""
+	for i := 0; i < words.MaxTokenLengthDefault; i++ {
+		str += "a"
+	}
+	testCases := []struct{
+		input string
+		expected []string
+	}{
+		{
+			input: str + ".aaa",
```

Comment:
What if we use `str[:len(str)-1] + ".aaa"`? What is supposed to happen? I forget.

Reply:
Hmm, let me see what Solr would do.